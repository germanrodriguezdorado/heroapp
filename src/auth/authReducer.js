import { types } from '../types/types';

// const state = {
//     name: "German",
//     logged: true
// }


export const authReducer = (state = {}, action) => {
    
    switch (action.type) {
        case types.login:
            // console.log(action)
            return {
                ...action.payload,
                logged: true
            }
        
        case types.logout:
            return {
                logged: false
            }
          
        default:
            break;
    }

}
