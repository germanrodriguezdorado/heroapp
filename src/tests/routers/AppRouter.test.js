import { shallow, mount } from "enzyme";
import { AuthContext } from "../../auth/AuthContext";
import { AppRouter } from "../../routers/AppRouter";



describe('Pruebas en componente AppRuter', () => {

    const contextValue = {
        dispatch: jest.fn(),
        user:{
            logged: false
        }
    };
   
    test('Debe de mostrar el login si no estoy authentincado', () => {
        
        const wrapper = mount(
            <AuthContext.Provider value={contextValue}>
                <AppRouter/>
            </AuthContext.Provider>
        );

        expect( wrapper ).toMatchSnapshot();

    });

    test('Debe de mostrar el componente Marvel si estÃ¡ autenticado', () => {

        const contextValue = {
            dispatch: jest.fn(),
            user: {
                logged: true
            }
        };

        const wrapper = mount(
            <AuthContext.Provider value={contextValue}>
                <AppRouter />
            </AuthContext.Provider>
        );

        expect(wrapper.find(".navbar").exists()).toBe(true);

        
    });


});
