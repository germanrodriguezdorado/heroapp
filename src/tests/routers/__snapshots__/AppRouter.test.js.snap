// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en componente AppRuter Debe de mostrar el login si no estoy authentincado 1`] = `
<div>
  <div
    className="container mt-5"
  >
    <h1>
      Welcome to HeroApp
    </h1>
    <hr />
    <button
      className="btn btn-primary"
      onClick={[Function]}
    >
      Login
    </button>
  </div>
</div>
`;
